### PR TITLE
Bonnie 1376 code equivalence

### PR DESCRIPTION
**Pull request by edeyoung**
Integrates cql-execution-engine updates from this PR: https://github.com/cqframework/cql-execution/pull/6. This makes the engine appropriately compare codes for equivalency and check code systems even if a code system is null.

Additionally, makes patient characteristics return codes **that include code systems** to work appropriately with the cql-execution-engine changes.

=====

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [ ] JIRA ticket for this PR:
- [ ] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.

**bonnie-v2.1**

```
=============================== Coverage summary ===============================
Statements   : 63.3% ( 1359/2147 )
Branches     : 33.66% ( 239/710 )
Functions    : 50.94% ( 245/481 )
Lines        : 60.03% ( 1143/1904 )
================================================================================
```

**bonnie**

```
=============================== Coverage summary ===============================
Statements   : 63.3% ( 1359/2147 )
Branches     : 33.66% ( 239/710 )
Functions    : 50.94% ( 245/481 )
Lines        : 60.03% ( 1143/1904 )
================================================================================
```

- [ ] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Comment by mayerm94** [comment]
Regression test is clean using updated cql4browsers.js

File: app/assets/javascripts/datatypes/patient_characteristic_ethnicity.js.coffee [hunk 1]
```diff
@@ -18,6 +18,8 @@ class CQL_QDM.PatientCharacteristicEthnicity extends CQL_QDM.QDMDatatype
 
   getCode: ->
     if @_patient?.has('ethnicity')
-      code: @_patient.get('ethnicity')
+      # NOTE: hard coding the code system. this will be removed when we switch
+      # to the QDM patient model
+      new cql.Code(@_patient.get('ethnicity'), 'CDC Race')
```
Comment (mayerm94) [inline app/assets/javascripts/datatypes/patient_characteristic_ethnicity.js.coffee]: I know it's only temporary, but VSAC lists the code_system as 'CDCREC', same for the race characteristic

Comment (edeyoung) [reply app/assets/javascripts/datatypes/patient_characteristic_ethnicity.js.coffee]: Good catch... in the execution engine comparison, it's 'CDC Race'. I will investigate...

Comment (edeyoung) [reply app/assets/javascripts/datatypes/patient_characteristic_ethnicity.js.coffee]: After doing some investigating and talking with a few people, we've decided to keep it as 'CDC Race'. We will transition to 'CDCREC' as we move towards the updated patient/measure/vsac models.

**Comment by edeyoung** [comment]
Updated w @losborne's test change recommendation.

After this has approvals, will tell @cmoesel to merge the cql-execution.
Will then update this with the latest cql4browsers from that.
Will then request a final regression check.
After final regression check passes, I will merge this in.
